src/core/public/chrome/ui/header/right_navigation_button.test.tsx

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { fireEvent, render } from '@testing-library/react';
+import { RightNavigationButton } from './right_navigation_button';
+import { applicationServiceMock, httpServiceMock } from '../../../../../core/public/mocks';
+
+const mockProps = {
+  application: applicationServiceMock.createInternalStartContract(),
+  http: httpServiceMock.createStartContract(),
+  appId: 'app_id',
+  iconType: 'mock_icon',
+  title: 'title',
+};
+
+describe('Right navigation button', () => {
+  it('should render base element normally', () => {
+    const { baseElement } = render(<RightNavigationButton {...mockProps} />);
+    expect(baseElement).toMatchSnapshot();
+  });
+
+  it('should call application getUrlForApp and navigateToUrl after clicked', () => {
+    const navigateToUrl = jest.fn();
+    const getUrlForApp = jest.fn();
+    const props = {
+      ...mockProps,
+      application: {
+        ...applicationServiceMock.createInternalStartContract(),
+        getUrlForApp,
+        navigateToUrl,
+      },
+    };
+    const { getByTestId } = render(<RightNavigationButton {...props} />);
+    const icon = getByTestId('rightNavigationButton');
+    fireEvent.click(icon);
+    expect(getUrlForApp).toHaveBeenCalledWith('app_id', {
+      path: '/',
+      absolute: false,
+    });
+    expect(navigateToUrl).toHaveBeenCalled();
```

Comment:
Shall we expect called with specific URL instead of getUrllForApp?

Reply:
Logic of generating has been moved to useMemo.

Reply:
Yes, logic of generating has been moved to useMemo.